### PR TITLE
add error handling, tracking for vault without purchase

### PR DESCRIPTION
**Pull request by wsbrunson**
### Summary
* moved `tests/standalone-vault.test.js` -> `src/card/interface/vault-without-purchase.test.js`
* added config to make vitest pick up ^^^ test, jest ignore it, and prettier it on commit
* added error handling for the vault without purchase flow
* added success and failure logging functions and invoked them at the appropriate time

The nesting is kinda hard to follow in `savePaymentSource` so open to suggestions. at a high-level

```
- createVaultSetupToken
- - getVaultSetupToken
- - updateVaultSetupToken
- - onApprove
- - log success
- - catch errors
- catch errors
```
this is to make available the `vaultSetupToken` from `createVaultSetupToken` to the rest of the flow

### CODE REVIEW

File: src/card/logger.js [hunk 1]
```diff
@@ -90,3 +90,34 @@ export function setupCardLogger({
...
+
+export const vaultWithoutPurchaseFailure = ({
+  vaultToken,
+  error
+}: {|
+  vaultToken?: string,
+  // should be Error but other apis are constraining this type
+  error: mixed
+|}) => {
+  const payload = {
```
Comment (siddy2181) [inline src/card/logger.js:112]: Similarly for this.
Should we move it to where the error will be thrown?
[R52-R55](https://github.com/paypal/paypal-smart-payment-buttons/pull/553/files#diff-30ce4e213d5295a3a447f401bd46ec120ebd6d205b3e6b312d38b0ac0ce44e4bR52-R55)

File: src/card/logger.js [hunk 2]
```diff
@@ -90,3 +90,34 @@ export function setupCardLogger({
     logger.flush();
   });
 }
+
+export const vaultWithoutPurchaseSuccess = ({
+  vaultToken
+}: {|vaultToken: string|}) => {
+  getLogger().track({
```
[the remark this answers is not in the document]
Comment (wsbrunson) [reply src/card/logger.js:97]: One of the reasons I did that was to make testing easier. Right now these are just `getLogger().track` but soon we will add `.info` and `.error` as well. Now we can just mock `vaultWithoutPurchaseSuccess` instead of trying to mock `getLogger()` and then all the sub-calls of `getLogger`

It also has an added benefit of keeping all of our logging calls in one place so if someone asks us to change anything about our logging calls, we know exactly where to go.

Does that change your opinion?

[the remark this answers is not in the document]
Comment (siddy2181) [reply src/card/logger.js:97]: Yes it does and that sounds good!!!

**Review by siddy2181 — APPROVED** [review]
